src/main/java/org/folio/event/handler/ItemCreateAsyncRecordHandler.java

```diff
@@ -0,0 +1,121 @@
+package org.folio.event.handler;
+
+import static org.folio.event.util.KafkaEventUtil.extractTenantFromHeaders;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import java.util.List;
+import java.util.Objects;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.ObjectUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.event.DomainEventPayloadType;
+import org.folio.rest.jaxrs.model.Piece;
+import org.folio.rest.persist.DBClient;
+import org.folio.services.piece.PieceService;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class ItemCreateAsyncRecordHandler extends BaseAsyncRecordHandler<String, String> {
+
+  private static final Logger log = LogManager.getLogger();
+
+  @Autowired
+  private PieceService pieceService;
+
+  public ItemCreateAsyncRecordHandler(Context context, Vertx vertx) {
+    super(vertx, context);
+    SpringContextUtil.autowireDependencies(this, context);
+  }
+
+  @Override
+  public Future<String> handle(KafkaConsumerRecord<String, String> kafkaConsumerRecord) {
+    if (log.isDebugEnabled()) {
```

Comment:
Probably don't need this check as message evaluation is not expensive, only calling `kafkaConsumerRecord.value()`

Reply:
removed, thanks